Add junit-gtest sample

fsladkey opened the pull request:
Hi there!

As mentioned in #382, there aren't currently any samples (here or elsewhere) for how to use the `androidx-test-ext:junit-gtest` library, so I've put together a minimal sample project.

This is not a particularly realistic example, but hopefully it will still be helpful as a guide for wiring everything together and could perhaps be expanded on in the future.

I based this on `ui/screenshot/ScreenshotSample`, to avoid deviating too far from the other samples. But definitely let me know if there's anything I should fix.

I'm not sure the directory placement is appropriate, I made a new directory called "native" because I wasn't sure the best way to categorize this sample (it's running unit tests, but on device 🤔). So just let know if there is a better spot for it.

Hope this is helpful! Thanks! :)

utzcoz commented:
@fsladkey Thanks for adding this sample for junit-gtest. It's awesome. Hi @brettchabot , what about helping to review this PR?

utzcoz commented on native/GTestExample/app/build.gradle:
```diff
@@ -0,0 +1,57 @@
+apply plugin: 'kotlin-android'
+
+android {
+    compileSdk 31
```
What about using sdk 32?

fsladkey replied:
Sure! Done.

utzcoz commented on native/GTestExample/app/src/androidTest/java/com/example/android/testing/nativesample/AdderTest.kt:
```diff
@@ -0,0 +1,9 @@
+import org.junit.runner.RunWith
+
+@RunWith(GtestRunner::class)
+@TargetLibrary(libraryName = "adder-test")
```
What about adding a Java version sample too?

fsladkey replied:
That's a good idea, unfortunately there's currently a limitation where only one `GtestRunner`  can exist at a a time so duplicating the test in Java throws an error. It would be great to remove this limitation and I'm hoping to but until then I'm not sure there's a clean way to have both.

I'd also be happy to _replace_ the Kotlin sample with Java is you think that's better. They are nearly identical anyway.

fsladkey commented on native/GTestExample/.gitignore:
```diff
@@ -0,0 +1,16 @@
+.gradle
+/local.properties
+/.idea/caches
+/.idea/libraries
```
Good idea, done!

fsladkey commented on native/GTestExample/gradle.properties:
```diff
@@ -0,0 +1,21 @@
+# https://developer.android.com/topic/libraries/support-library/androidx-rn
+android.useAndroidX=true
+# Automatically convert third-party libraries to use AndroidX
+android.enableJetifier=true
```
Yep that was just copied in from the other sample, it seems to be fine without.

fsladkey commented:
Thanks for the reviews folks! I think I addressed most of the comments and added a readme. I also tested the `./test_all.sh` script and although I had to comment out some other projects (maybe something in my environment is wrong) it looks like this project should now be tested by that script and is passing.

DanAlbert commented on unit/BasicNativeAndroidTest/app/build.gradle:
```diff
@@ -0,0 +1,69 @@
+}
+apply plugin: 'kotlin-android'
+
+android {
```
I haven't gotten around to filing a bug for this, but consumers of junit-gtest actually need to exclude some cruft that's included in the AAR or it will end up in the APK:  https://github.com/DanAlbert/ndk-app-template/blob/6f46d2f49e68d20c102eb2122262716564f191d2/app/build.gradle#L42-L48 (in this example you don't want to exclude libc++_shared since this build is using libc++_shared; the example I linked uses the static lib)

Not sure if you want to include that workaround in this example, or fix the bug before submitting this. I'm fine either way.

The rest LGTM though.
